data-prepper-plugins/geoip-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/GeoIPProcessor.java

```diff
@@ -79,34 +92,62 @@ public GeoIPProcessor(final PluginMetrics pluginMetrics,
   public Collection<Record<Event>> doExecute(final Collection<Record<Event>> records) {
     Map<String, Object> geoData;
 
+    final GeoIPDatabaseReader geoIPDatabaseReader = geoIPProcessorService.getGeoIPDatabaseReader();
+    final boolean databasesExpired = geoIPDatabaseReader.isExpired();
+
     for (final Record<Event> eventRecord : records) {
       final Event event = eventRecord.getData();
+
       final String whenCondition = geoIPProcessorConfig.getWhenCondition();
 
       if (whenCondition != null && !expressionEvaluator.evaluateConditional(whenCondition, event)) {
         continue;
       }
+      geoIpEventsProcessed.increment();
+
+      // TODO: Need to decide the behaviour, right now if all databases are expired we don't enrich the data.
```

Comment:
I think just having this processor be a no-op on Events is ok when the database is expired, but we should be sure to tag these Events so that they can be routed appropriately in downstream sink and processors. I would say we could even default the tag to be `_expired_geoip_database`, and could make it configurable with `tags_on_expired_database` or something along those lines, given that this is one of the 2 special cases of errors

Reply:
Yeah there will be tags added if database expired. The default tags are not being added right now and we will consider a mechanism on how we can add tags in plugins and get back to it. Right now it will be configurable